src/public/media.ts

```diff
@@ -407,4 +413,66 @@ export namespace media {
     const messageId = sendMessageRequestToParent('viewImages', params);
     GlobalVars.callbacks[messageId] = callback;
   }
+
+  /**
+   * Barcode configuration supplied to scanBarCode API to customize barcode scanning experience in mobile
+   * All properties in BarCodeConfig are optional and have default values in the platform
+   */
+  export interface BarCodeConfig {
+    /**
+     * Optional; Lets the developer specify the type of barcode to be scanned
+     * Default value is All i.e both one dimensional and two dimensional(only QRcode) barcodes can be scanned
+     */
+    barCodeType?: BarCodeType;
+    /**
+     * Optional; Lets the developer specify the scan timeout interval in seconds
+     * Default value is 30 seconds and max allowed value is 60 seconds
+     */
+    timeOutIntervalInSec?: number;
+  }
+
+  /**
+   * Specifies the types of barcode which are supported by scanBarCode API
+   */
+  export enum BarCodeType {
+    /**
+     * One dimensional barcode types and QRcode are supported
+     */
+    All = 1,
```

Comment:
How would this work if we add more types in the future? For example, what if a ThreeDimensional option was added and the developer wanted to specify only TwoDimensional and ThreeDimensional scanning? Seems like these should either be modeled as flags or we should just change `barCodeType` to a `barCodeTypes` array in the BarCodeConfig interface allowing the caller to provide multiple options.

Reply:
Good catch but we have internally discussed in the team on keeping the optional barcodeType field and finally decided to remove it from BarCodeConfig. Currently, we don't see much value in keeping the optional barCodeType field in the barcodeConfig as Office Lens barcode SDK is only supporting 1 D barcode types and QRcode from 2 Dimension barcode types. We can introduce barCodeType as an array in future if more barcode types are supported by Office Lens and there is an ask from customers to support custom barcode type scanning.